Add new Style Guide template to incubator (#4)
thegooddocsproject/incubator: style-guide/template-style-guide.md

> <!-- Copy this Template. -->

Reviewer: What we discovered in our first round of templates was that any comment (like this first line) are not visible to people who view the template from github.
As such, I recommend using another convention. As you are already using curly brackets, maybe use italic wrapped by square brackets. (Maybe also use italic for what is inside curly brackets too.)

Author: Sounds good. I've removed any comments and replaced them with {placeholder text}.

strives to communicate respectfully to all people. This kind of language is
sensitive to differences and seeks to promote equal opportunities.

For resources on making your writing more inclusive, see:

- [Inclusive documentation - Google developer’s style guide](https://developers.google.com/style/inclusive-documentation)
- [The Conscious Style Guide - a collection of resources](https://consciousstyleguide.com/)
- [Bias-free communication - Microsoft Writing Style Guide](https://docs.microsoft.com/en-us/style-guide/bias-free-communication)
- [Guidelines for Inclusive Language - Linguistic Society of America](https://www.linguisticsociety.org/resource/guidelines-inclusive-language)

## Using linters

<!-- This section is optional -->

This project uses the {your preferred linter.}

<!-- Provide instructions or policies related to the linter here. -->

## How the style guide is updated

<!-- Indicate here how frequently your style guide is reviewed, who owns the
style guide, and how contributors can provide feedback on your style guide. -->

## Revision history

<!-- This section is optional -->

The following table describes the history of all decisions and revisions made to

> this style guide over time. For style guide drafts, use 0.1 through 0.9. After

Reviewer: I suggest add (and possibly replace):
This document follows the Major.Minor.Patch [semantic versioning convention](https://semver.org/).

Author: That's a good suggestion too. I've applied your wording to the style guide template and then moved the old text to the style guide description because it seemed more appropriate there anyhow.

for example     | e.g.               |  Avoid non-English words
people with disabilities  | <ul><li>the disabled</li><li>disabled people</li><li>people with handicaps</li><li>the handicapped</li></ul>  |  Use inclusive and bias-free language. See [Accessible Writing](#accessible-writing
that is         |  i.e.              |  Avoid non-English words

## Topic Types and Templates

This project recommends using the following templates from the
[Good Docs project](https://github.com/thegooddocsproject/templates):

- API Overview
- API Quickstart
- API Reference
- Discussion Article
- How To Guide
- Logging Article
- Reference Article
- Tutorial

## General Writing Tips

<!-- This section is optional -->

For some general tips about improving writing see:

- [Write the Docs - Style Guides](https://www.writethedocs.org/guide/writing/style-guides/#writing-style)
- [18F Content Guide](https://content-guide.18f.gov/)

## Accessible Writing

Documentation should be written in a way that supports people with disabilities
and users with various input methods and devices. Improving accessibility also
helps make documentation clearer and more useful for everyone.

> For resources on making your writing more accessible, see:

Reviewer 2: I've made a few suggestions to use the linked resources' own titles and capitalization. I think diverging on this sort of thing can lead to needless confusion (e.g., the Microsoft Writing Style Guide is the successor to the Manual of Style, which is no longer published). Plus, it's easier to tell I've landed on the right page.

Because there were a few of them in one place, I flagged them here, but some of these get used more than once and I didn't trust I'd get them all. You'll want to check throughout for each of these.

Author: I realized I hadn't entirely fixed this issue that you raised, so I've gone through and cleaned up all the references to Google developer documentation style guide and Microsoft Writing Style Guide now. I checked any references to other sources should now be correct with my latest commit.

Reviewer 2: Oh, good catch. I forgot to recheck this myself. Thank you!

> - [Writing accessible documentation - Google Developer’s Style Guide](https://developers.google.com/style/accessibility)

Reviewer 2: ```suggestion
- [Writing accessible documentation - Google developer documentation style guide](https://developers.google.com/style/accessibility)
```

Author: Good catch! I've updated it in a few more places as well.

> - [Accessibility guidelines and requirements - Microsoft Manual of Style](https://docs.microsoft.com/en-us/style-guide/accessibility/accessibility-guidelines-requirements)

Reviewer 2: ```suggestion
- [Accessibility guidelines and requirements - Microsoft Writing Style Guide](https://docs.microsoft.com/en-us/style-guide/accessibility/accessibility-guidelines-requirements)
```

Author: Another good catch. I've updated this in multiple places.

> - [Writing for Accessibility - MailChimp Content Style Guide](https://styleguide.mailchimp.com/writing-for-accessibility/)

Reviewer 2: ```suggestion
- [Writing for Accessibility - Mailchimp Content Style Guide](https://styleguide.mailchimp.com/writing-for-accessibility/)
```

Author: Thanks for double-checking the camel case!

who contributes to the project’s documentation.

## Intended audience and scope

This style guide is intended for use by any contributors that are writing
documentation for {Your Project Name}, including software engineers. This guide
can help project contributors to communicate clearly and consistently in {you
define the scope: the project's end-user documentation, API documentation,
forum posts, and so forth.}

## Our preferred style guide

We have adopted the
{[your preferred style guide, such as the Google developer's style guide](https://developers.google.com/style)}
for {Your Project} documentation. For a quick summary, see the
[Google style guide highlights](https://developers.google.com/style/highlights).
The rest of this document describes our project-specific customizations to
{Google's or your preferred} guide.

Our project uses {your choice: standard American spelling or standard British
spelling, etc.} and our preferred dictionary is the
{[American Heritage Dictionary](https://ahdictionary.com/) or your preferred
dictionary}.

When writing documentation for our project, align with the default guide’s
voice and tone.

## Glossary of preferred terms

> {This first paragraph is optional or you could include in the word list.}

Reviewer 3: ```suggestion
{Calling out how to write your project name first is optional. You can choose to include your project name in the word list instead.}
```
I had to read this several times before I understood what it meant.

strives to communicate respectfully to all people. This kind of language is
sensitive to differences and seeks to promote equal opportunities.

For resources on making your writing more inclusive, see:

- [Inclusive documentation - Google developer’s style guide](https://developers.google.com/style/inclusive-documentation)
- [The Conscious Style Guide - a collection of resources](https://consciousstyleguide.com/)
- [Bias-free communication - Microsoft Writing Style Guide](https://docs.microsoft.com/en-us/style-guide/bias-free-communication)
- [Guidelines for Inclusive Language - Linguistic Society of America](https://www.linguisticsociety.org/resource/guidelines-inclusive-language)

## Using linters

{This section is optional.}

This project uses the {your preferred linter.}

{Provide instructions or policies related to the linter here.}

## How the style guide is updated

{Indicate here how frequently your style guide is reviewed, who owns the style
guide, and how contributors can provide feedback on your style guide.}

## Revision history

{This section is optional or can be combined with the next section if needed.}

> The following table describes the history of all decisions and revisions made to

Reviewer 3: ```suggestion
This table describes the history of all decisions and revisions made to
```

Author: I'm not going to adopt this suggestion because I prefer my wording. It's also the convention throughout the document, so if I were to adopt this suggestion, I'd need to implement it in multiple places.

Reviewer 3: NP.
